src/io/export_pov.cc

```diff
@@ -0,0 +1,122 @@
+ *  (at your option) any later version.
+ *
+ *  As a special exception, you have permission to link this program
+ *  with the CGAL library and distribute executables, as long as you
+ *  follow the requirements of the GNU GPL in regard to all of the
+ *  software in the executable aside from CGAL.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ */
+
+#include "export.h"
+
+#include "PolySet.h"
+#include "PolySetUtils.h"
+
+
+void export_pov(const std::shared_ptr<const Geometry>& geom, std::ostream& output, const ExportInfo& exportInfo)
+{
+  auto ps = PolySetUtils::getGeometryAsPolySet(geom);
+  if (Feature::ExperimentalPredictibleOutput.is_enabled()) {
+    ps = createSortedPolySet(*ps);
+  }
+
+  output << "// Generated by OpenSCAD!\n";
+  output << "// Source file: " << exportInfo.sourceFileName << "\n\n";
+
+  output << "#version 3.7;\n";
+  output << "global_settings { assumed_gamma 1.0 }\n";
+  output << "#declare MATERIAL=finish { specular 0.5 roughness 0.001 reflection{0 0.63 fresnel} ambient 0 diffuse 0.6 conserve_energy }\n";
+  output << "#declare MATERIAL_INT=interior{ior 1.32}\n";
+
+  auto has_color = !ps->color_indices.empty();
+
+  double avg_x = 0;
+  double avg_y = 0;
+  double avg_z = 0;
+  double tsd_x = 0;
+  double tsd_y = 0;
+  double tsd_z = 0;
+  size_t avg_n = 0;
+
+  for (size_t pi=0; pi<ps->indices.size(); pi++) {
+    const auto &t = ps->indices.at(pi);
+    output << "polygon { " << t.size() + 1 << ", \n";
+    for (size_t i=0; i<t.size(); i++) {
+      if (i)
+        output << ", ";
+      output << "<" << ps->vertices[t.at(i)].x() << ", " << -ps->vertices[t.at(i)].y() << ", " << ps->vertices[t.at(i)].z() << ">";
+      avg_x += ps->vertices[t.at(i)].x();
+      avg_y += -ps->vertices[t.at(i)].y();
+      avg_z += ps->vertices[t.at(i)].z();
+      tsd_x += ps->vertices[t.at(i)].x() * ps->vertices[t.at(i)].x();
+      tsd_y += ps->vertices[t.at(i)].y() * ps->vertices[t.at(i)].y();
+      tsd_z += ps->vertices[t.at(i)].z() * ps->vertices[t.at(i)].z();
+      avg_n++;
+    }
+    double r = 1.0, g = 1.0, b = 1.0, a = 0.0;
+    if (has_color) {
+      auto color_index = ps->color_indices[pi];
+      if (color_index >= 0) {
+        auto color = ps->colors[color_index];
+        r = std::clamp(double(color[0]), 0., 1.);
+        g = std::clamp(double(color[1]), 0., 1.);
+        b = std::clamp(double(color[2]), 0., 1.);
+        a = std::clamp(1.0 - color[3], 0., 1.);
+      }
+    }
+    if (r == 0 && g == 0 && b == 0)  // work around for black objects in 5185
```

Comment:
Is this a workaround for #5216 ? If so, I suggest leaving it unfixed and add a test case.

Reply:
This probably still needs addressing.